debugger/pythonExtension/gdb_sgx_plugin.py

```diff
@@ -77,8 +77,8 @@ def is_valid(self):
         return self.magic == self.MAGIC_VALUE
 
 
-# This constant definition must align with the OE enclave layout.
-TD_OFFSET_FROM_TCS =  0X4000
+# This constant definition must align with sgx_tcs_t
+TCS_GSBASE_OFFSET =  0X38
```

Comment:
Maybe we should change this to an integer since this represents a byte offset in the sgx_tcs_t struct. When I first went through this I was very confused because 38 didn't seem to add up to get to the `gsbase` element, and then I realized this number is hexadecimal for 56! It was my mistake of course, I just usually associate byte offsets with decimal numbers! 😊

Reply:
C/C++ have things like the offsetof() macro (https://en.cppreference.com/w/cpp/types/offsetof).
Is there anything usable in Python rather than hardcoding a possibly fragile number?

Reply:
@johnkord Nice suggestion. Done.

Reply:
@dthaler There is nothing readily available in python to calculate the offset of a C struct's field. 

Though the offset is hard-coded, the accompanying test asserts that the hard-coded offset is equal to the offset determined by the C compiler. Offset mismatch would fail the test.

In the future, these offsets will be part of the [debugger binary contract](https://github.com/openenclave/openenclave/blob/master/include/openenclave/internal/debugrt/host.h) which is a versioned contract between the debugger and the SDK. Most of the debugger has already been moved to the contract; just a few pending pieces we are trying to address as per scheduling.